app.rb

```diff
@@ -0,0 +1,158 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  MENU_OPTIONS = {
+    '1' => :list_books,
+    '2' => :list_people,
+    '3' => :create_person,
+    '4' => :create_book,
+    '5' => :create_rental,
+    '6' => :list_all_rentals,
+    '7' => :exit_program
+  }.freeze
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def start
+    loop { choose_option }
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books available. Please add some books and try again.'
+    else
+      puts "There are #{@books.count} book(s) available."
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people added yet. Please add a person and try again.'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index + 1}) [#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Would you like to create a student (1) or a teacher (2)? Select a number: '
+    choice = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(age, name)
+    when '2'
+      create_teacher(age, name)
+    else
+      puts 'Invalid choice. Please try again.'
+    end
+  end
+
+  def create_student(age, name)
+    print 'Grade: '
+    grade = gets.chomp
+    print 'Parent permission? [Y/N]: '
+    permission_option = gets.chomp.downcase
+    student = Student.new(grade, age, name, parent_permission: permission_option == 'y')
+    @people << student
+    puts "Student created successfully. ID is #{student.id}"
+  end
```

Comment:
Example video
![image](https://github.com/melkamu12/OOP-school-library/assets/78251927/4257ebfa-d515-4de0-893b-686f106afd62)

You app
![image](https://github.com/melkamu12/OOP-school-library/assets/78251927/16be4479-b316-4c91-a06f-f6d418a97d22)

It is important according to the project requirements that our school library console app behaves in the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) it was provided. 😊

In the video, to create a student we only require 3 parameters:
`Age`, `Name`, and `Has parent permission?`

Please correct that 💪🛠